ui/StatusQ/src/StatusQ/Components/StatusRoundedMedia.qml

```diff
@@ -0,0 +1,133 @@
+import QtQuick 2.15
+import QtQml 2.15
+import QtGraphicalEffects 1.0
+import QtMultimedia 5.15
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+StatusRoundedComponent {
+    id: root
+
+    enum MediaType {
+        Image,
+        Video,
+        Web,
+        Unknown
+    }
+
+    property url mediaUrl
+    property string mediaType
+    readonly property int componentMediaType: {
+        if (root.mediaType.startsWith("image")) {
+            return StatusRoundedMedia.MediaType.Image
+        } else if (root.mediaType.startsWith("video")) {
+            return StatusRoundedMedia.MediaType.Video
+        } else if (root.mediaType.startsWith("text/html")) {
+            return StatusRoundedMedia.MediaType.Web
+        }
+        return StatusRoundedMedia.MediaType.Unknown
+    }
+
+    property url fallbackImageUrl
+
+    isLoading: {
+        if (mediaLoader.status === Loader.Ready) {
+            return mediaLoader.item.isLoading
+        }
+        return true
+    }
+/*
+    readonly property bool isError: {
+        if (mediaLoader.status === Loader.Ready) {
+            return mediaLoader.item.isError
+        }
+        return true
+
+    }
+*/
+    QtObject {
+        id: d
+        property bool isError: false
+        property bool isFallback: false
+        property int errorCounter: 0
+        Binding on isError {
+            when: mediaLoader.status === Loader.Ready
+            value: mediaLoader.item ? mediaLoader.item.isError : true
+            delayed: true
+            restoreMode: Binding.RestoreBindingOrValue
+        }
+
+        onIsErrorChanged: {
+            if (isError) {
+                errorCounter = errorCounter + 1
+                root.processError()
+            }
+        }
+    }
+
+    Loader {
+        id: mediaLoader
+        anchors.fill: parent
+    }
+
+    Component.onCompleted: updateMediaLoader()
+    onMediaUrlChanged: updateMediaLoader()
+    onComponentMediaTypeChanged: updateMediaLoader()
+
+    function updateMediaLoader() {
+        if (root.mediaUrl !== "") {
+            if (componentMediaType === StatusRoundedMedia.MediaType.Image) {
+                mediaLoader.setSource("StatusAnimatedImage.qml",
+                                    {
+                                        "source": root.mediaUrl
+                                    });
+                return
+            } else if (componentMediaType === StatusRoundedMedia.MediaType.Video) {
+                mediaLoader.setSource("StatusVideo.qml",
+                                    {
+                                        "player.source": root.mediaUrl
+                                    });
+                return
+            } else if (componentMediaType === StatusRoundedMedia.MediaType.Web) {
+                mediaLoader.setSource("StatusWebEngineView.qml",
```

Comment:
Uhm... do we really want to embed a (potentially insecure) web browser view here?

Reply:
Figured this might raise a couple red flags... Will remove it for now, it was basically free after implementing the other types.